### PR TITLE
add instructions for ssl certificate creation

### PR DESCRIPTION
**Pull request by lindawoo-plaid**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -109,11 +109,49 @@ Upon receipt of a transactions webhook a call will be made to Plaid's transactio
...
 
-You will also need to configure `http://localhost:3001/oauth-link` as an allowed redirect URI for your client ID through the [Plaid developer dashboard](https://dashboard.plaid.com/team/api).
+To test the OAuth flow in sandbox, choose 'Playtypus OAuth Bank' from the list of financial institutions in Plaid Link.
 
-To test the OAuth flow, choose 'Playtypus OAuth Bank' from the list of financial instutions in Plaid Link.
+#### Instructions for using https with localhost
+
+If you want to test OAuth in development, you need to use https and set `PLAID_REDIRECT_URI=https://localhost:3001/` in `.env`. In order to run your localhost on https, you will need to create a self-signed certificate and add it to the client root folder. You can use the following instructions to do this. However, please note that this is recommended for local development environments only and not for production environments.
```
Comment (phoenixy1) [inline README.md:118]: i thought the redirect uri was https://localhost:3001/oauth-link?

**Review by phoenixy1 — COMMENTED** [review]
These instructions don't work for me on Windows and I strongly suspect they wouldn't have worked for me on the Quickstart either if I had been using Docker on Windows. I get the following error:

You specified SSL_CRT_FILE in your env, but the file "/opt/client/localhost.pem" can't be found

As a broader issue, given all of the problems we've had with Docker on non-Mac setups, combined with the change to Docker desktop's pricing structure making it much less attractive to users at larger companies, I strongly think we should set aside some time during reset week and/or Q2 to investigate options for moving away from Docker for these example apps.

https://github.com/plaid/pattern/issues/219

File: README.md (hunk 1 above)
Comment (lindawoo-plaid) [reply README.md:118]: oops...you are right!

**Comment by Tarious1** [comment]
yes. you may merge them and I have verified my bank that's linked to plaid.  please send access thanks keep me informed

File: README.md [hunk 2]
```diff
@@ -109,11 +109,64 @@ Upon receipt of a transactions webhook a call will be made to Plaid's transactio
...
+  "start": "PORT=3001 react-scripts start",
+```
+
+with this line instead:
+
+```bash
+"start": "PORT=3001 HTTPS=true react-scripts start",
+```
+
+You may still get an invalid certificate warning on your browser. If so, click on "advanced" and proceed. Also on Windows, the frontend may still try to load http://localhost:3001 and you may have to access https://localhost:3001 manually.
```
Comment (phoenixy1) [inline README.md:169]: personally for me on windows the frontend doesn't automatically load at all

**Review by phoenixy1 — DISMISSED** [review]
will these changes also be applied to the pattern account funding app?

**Comment by Tarious1** [comment]
Linda, I don't understand.

**Comment by Tarious1** [comment]
> 

f9f2f6517a3f3f96eb33331bc396405cd39314af

File: README.md (hunk 2 above)
Comment (lindawoo-plaid) [reply README.md:169]: actually on Docker it doesn't load at all, so i will remove this line.

**Comment by lindawoo-plaid** [comment]
Yes, the changes will be applied to the account-funding app as well.

File: README.md [hunk 3]
```diff
@@ -109,11 +109,88 @@ Upon receipt of a transactions webhook a call will be made to Plaid's transactio
...
+while [ "$(curl -s -o /dev/null -w "%{http_code}" -m 1 localhost:3001)" != "200" ]
+```
+
+with this line instead:
+
+```bash
+while [ "$(curl -s -o /dev/null -w "%{http_code}" -m 1 https://localhost:3001)" != "200" ]
+```
+
+After starting up the Pattern sample app, you can now view it at https://localhost:3001. Your browser will alert you with an invalid certificate warning on your browser; click on "advanced" and proceed.
```
Comment (phoenixy1) [inline README.md:193]:
```suggestion
After starting up the Pattern sample app, you can now view it at https://localhost:3001. Your browser will alert you with an invalid certificate warning; click on "advanced" and proceed.
```

**Review by phoenixy1 — APPROVED** [review]

**Review by phoenixy1 — APPROVED** [review]

**Comment by Tarious1** [comment]
Hi Linda,
 * I have received emails about issues with the workflow since June of last
year. Is this normal. what can be done to simplify this?*

**Comment by phoenixy1** [comment]
@Tarious1 Hi! For clarification, Plaid Pattern is provided by Plaid for demonstration and developer education purposes and Plaid employees cannot answer any questions or support inquiries related to its use in anything other than a demonstration or developer education capacity. If you have questions about a Plaid-powered app (including the use of a Pattern-based app in a Production deployment), please contact the developer of the app directly (not via this forum). If you would like to have these discussions with the developer via GitHub, you can request that they fork the GitHub repo so you can discuss the app there. Going forward, we will enforce this policy on our GitHub. Thanks!

**Pull request merged**